objects/src/accounts/account_id.rs

```diff
@@ -57,145 +69,196 @@ impl AccountId {
         }
 
         Ok(())
     }
 }
 
 impl Deref for AccountId {
-    type Target = [Felt; 3];
+    type Target = Felt;
 
     fn deref(&self) -> &Self::Target {
         &self.0
     }
 }
 
-impl From<AccountId> for [Felt; 3] {
+impl From<AccountId> for Felt {
     fn from(id: AccountId) -> Self {
         id.0
     }
 }
 
 impl From<AccountId> for Word {
     fn from(id: AccountId) -> Self {
-        [id.0[0], id.0[1], id.0[2], ZERO]
+        [ZERO, ZERO, ZERO, id.0]
     }
 }
 
-impl From<AccountId> for [u8; 24] {
+impl From<AccountId> for [u8; 8] {
     fn from(id: AccountId) -> Self {
-        let mut result = [0_u8; 24];
-        result[..8].copy_from_slice(&id.0[0].as_int().to_le_bytes());
-        result[8..16].copy_from_slice(&id.0[1].as_int().to_le_bytes());
-        result[16..].copy_from_slice(&id.0[2].as_int().to_le_bytes());
+        let mut result = [0_u8; 8];
+        result[..8].copy_from_slice(&id.0.as_int().to_le_bytes());
         result
     }
 }
 
-/// This conversion is possible because the 3 least significant bytes of an account ID are always
-/// set to zeros.
-impl From<AccountId> for [u8; 21] {
+impl From<AccountId> for u64 {
     fn from(id: AccountId) -> Self {
-        let mut result = [0_u8; 21];
-        result[..8].copy_from_slice(&id.0[0].as_int().to_le_bytes());
-        result[8..16].copy_from_slice(&id.0[1].as_int().to_le_bytes());
-        result[16..].copy_from_slice(&id.0[2].as_int().to_le_bytes()[..5]);
-        result
+        id.0.as_int()
     }
 }
 
-impl TryFrom<[Felt; 3]> for AccountId {
+impl TryFrom<Felt> for AccountId {
     type Error = AccountError;
 
-    fn try_from(value: [Felt; 3]) -> Result<Self, Self::Error> {
+    fn try_from(value: Felt) -> Result<Self, Self::Error> {
         let id = Self(value);
         id.validate()?;
         Ok(id)
     }
 }
 
-impl TryFrom<[u8; 24]> for AccountId {
+impl TryFrom<Word> for AccountId {
+    type Error = AccountError;
+
+    fn try_from(value: Word) -> Result<Self, Self::Error> {
+        debug_assert_eq!(value[..3], [ZERO, ZERO, ZERO]);
+        Self::try_from(value[3])
+    }
+}
+
+impl TryFrom<[u8; 8]> for AccountId {
     type Error = AccountError;
 
-    fn try_from(value: [u8; 24]) -> Result<Self, Self::Error> {
-        let elements =
-            [parse_felt(&value[..8])?, parse_felt(&value[8..16])?, parse_felt(&value[16..])?];
-        Self::try_from(elements)
+    fn try_from(value: [u8; 8]) -> Result<Self, Self::Error> {
+        let element = parse_felt(&value[..8])?;
+        Self::try_from(element)
     }
 }
 
-impl TryFrom<[u8; 21]> for AccountId {
+impl TryFrom<u64> for AccountId {
     type Error = AccountError;
 
-    fn try_from(value: [u8; 21]) -> Result<Self, Self::Error> {
-        let mut bytes = [0_u8; 24];
-        bytes[..21].copy_from_slice(&value);
-        Self::try_from(bytes)
+    fn try_from(value: u64) -> Result<Self, Self::Error> {
+        let element = Felt::from(value);
```

Comment:
We should make sure here that `value` is a valid field element (I think `Felt::from()` will do a silent conversion).